cmake/kokkos_cxx.cmake

```diff
@@ -56,7 +56,7 @@ IF(INTERNAL_HAVE_COMPILER_NVCC)
 
   STRING(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+$"
          TEMP_CXX_COMPILER_VERSION ${INTERNAL_CXX_COMPILER_VERSION})
-  SET(KOKKOS_CXX_COMPILER_VERSION ${TEMP_CXX_COMPILER_VERSION} CACHE STRING INTERNAL)
+  SET(KOKKOS_CXX_COMPILER_VERSION ${TEMP_CXX_COMPILER_VERSION} CACHE STRING INTERNAL FORCE)
```

Comment:
Fixes #2190.

Reply:
Please add a comment to capture why it is needed.